src/game.py

```diff
@@ -43,31 +44,55 @@ def process_user_input(self):
         self.total_marks_on_board = self.board.count_marks(self.game_board)
         return self.game_board
 
+    def new_game(self):
+        self.total_marks_on_board = 0
+        self.game_board = Board().starter_board
+        self.get_formatted_board()
+        self.play_game()
+
     def get_winning_mark(self, total_marks_on_board):
         player = self.get_current_player(total_marks_on_board)
         if player == self.player_one:
             return self.player_two
         else:
             return self.player_one
 
+    def handle_winning_game(self, board):
+        winner = self.get_winning_mark(self.total_marks_on_board)
+        message.display_winner_message(self, winner)
+        self.repeat_game()
+
+    def handle_draw(self):
+        self.get_prompt(self.total_marks_on_board)
+        self.process_user_input()
+        self.total_marks_on_board = self.board.count_marks(self.game_board)
+        self.get_formatted_board()
+        self.play_game()
+
+    def repeat_game(self):
+        message.display_play_again(self)
+        self.play_again = False
+        if self.user_interface.get_play_again_user_input() == "Y":
+            self.play_again = True
+        while self.play_again:
+            self.new_game()
+        else:
+            self.play_again = False
+
     def play_game(self):
         total_marks_on_board = self.board.count_marks(self.game_board)
-        while not self.board.determine_is_full(total_marks_on_board, self.game_board):
-            self.get_prompt(total_marks_on_board)
-            self.process_user_input()
-            total_marks_on_board = self.board.count_marks(self.game_board)
-            self.get_formatted_board()
+        while not self.board.is_full(total_marks_on_board, self.game_board):
             if self.rules.is_winner(self.game_board):
-                winner = self.get_winning_mark(total_marks_on_board)
-                message.display_winner_message(self, winner)
+                self.handle_winning_game(self.game_board)
                 break
             else:
-                self.play_game()
+                self.handle_draw()
                 break
         else:
             self.get_prompt(total_marks_on_board)
+            self.repeat_game()
 
     def run(self):
         message.display_welcome_message(self)
-        self.get_formatted_board()
-        self.play_game()
+        self.new_game()
+        message.display_goodbye_message(self)
```

Comment:
This kicks things off, but at the top level is where the game loop should be declared.  With the way things work at the moment, I am not sure what is happening now with all the nested calls.